services/src/main/java/org/keycloak/protocol/oidc/AccessTokenIntrospectionProvider.java

```diff
@@ -85,6 +85,13 @@ public Response introspect(String token) {
                         }
                     }
                 }
+
+                if (accessToken.getConfirmation() != null && accessToken.getConfirmation().getKeyThumbprint() != null) {
```

Comment:
In the description of the issue, I suggested also to add `DPoP` instead of `Bearer` as the token type during authentication - in case of Bearer tokens. Does this look ok to you?

Maybe the easiest would be:
```
tokenMetadata.put(OAuth2Constants.TOKEN_TYPE, accessToken.getType());
```

WDYT?

Reply:
I see, only `tokenMetadata.put(OAuth2Constants.TOKEN_TYPE, accessToken.getType());` is enough.

Reply:
I guess that when token type is updated to `DPoP` for the DPoP tokens, some small updates might be needed also in other places. I think that `TokenVerifier.withDefaultChecks()` may need to be updated to allow both `Bearer` and `DPoP` as token types.

Reply:
But maybe not... We can see if there are any failing tests :-)